### ISSUE TITLE
Markdown links are being removed

### ISSUE THREAD

**Issue by Dan503** [hosseinmd/prettier-plugin-jsdoc#158]
I recently upgraded from `v0.3.12` to `v0.3.33`.

In one of the versions after `v0.3.12` it has started removing the markdown links in my [Type-Doc](https://typedoc.org/guides/doccomments/) comments :(

**Input**

```ts
interface Test {
    /**
     * Name of something.
     *
     * See [documentation][1] for more details.
     *
     * [1]: https://www.documentation.com
     */
    name?: string
}
```

**Output** (Unwanted removal of link markdown)

```ts
interface Test {
    /**
     * Name of something.
     *
     * See documentation for more details.
     */
    name?: string
}
```

The output should be retaining the markdown link.

**Comment by hosseinmd** [comment]
That comment isn't md standard.
Change your comment to this

```
Name of something.
See [documentation](1) for more details.
# 1: https://www.documentation.com
```

**Comment by Dan503** [comment]
Are you sure?

Your syntax appears to be the non-standard one to me.

https://www.markdownguide.org/basic-syntax/#reference-style-links

Or if there is a flavour of markdown that works with your style, you should support this style.

**Comment by hosseinmd** [comment]
GitHub's comment is markdown test your link

**Comment by hosseinmd** [comment]
It is hard to support `Reference-style Links`. But I will work on it.

**Comment by Dan503** [comment]
This is a [test link][1] to test it works in GitHub

```
This is a [test link][1] to test it works in GitHub

[1]: https://www.google.com/
```

[1]: https://www.google.com/

**Comment by hosseinmd** [comment]
Ok,
That is  a `[test link](https://www.google.com/)`  to test it works in GitHub

is correct, and work in GitHub but  
This is a `[test link][1]` to test it works in GitHub
not working

**Comment by Dan503** [comment]
My comment here proves that it is accepted by GitHub.

https://github.com/hosseinmd/prettier-plugin-jsdoc/issues/158#issuecomment-1083796423

It is a reference-link-only syntax. `[test link][1]` on it's own doesn't work without a reference.

If you copy the sample text I posted in that comment you will see that GitHub does understand it.

**Comment by hosseinmd** [comment]
Fixed: v0.3.35

Thank you.

**Comment by Dan503** [comment]
I installed v0.3.35 into my project. it seems like `@param` might be breaking it.

**input**

```ts
/**
 * This is a [link][1] test.
 *
 * [1]: https://www.google.com/
 *
 * @param value - Any value you want to test is a number.
 */
function test(value) {
    return true
}
```

**current output**

```ts
/**
 * This is a link test.
 *
 * @param value - Any value you want to test is a number.
 */
function test(value) {
    return true
}
```

**expected output**

```ts
/**
 * This is a [link][1] test.
 *
 * [1]: https://www.google.com/
 *
 * @param value - Any value you want to test is a number.
 */
function test(value) {
    return true
}
```

**Comment by Dan503** [comment]
I installed v0.3.35 into my project.

It isn't quite providing the correct output.

**input**

```ts
/**
 * This is a [link][1] test.
 *
 * [1]: https://www.google.com/
 *
 * @param value - Any value you want to test is a number.
 */
function test(value) {
    return true
}
```

**current output**

```ts
/**
 * This is a [link][1] test.
 *
 * [1]: [https://www.google.com/]
 *
 * @param value - Hello
 */
function test(value) {
    return true
}
```

**expected output**

```ts
/**
 * This is a [link][1] test.
 *
 * [1]: https://www.google.com/
 *
 * @param value - Any value you want to test is a number.
 */
function test(value) {
    return true
}
```

Notice that it is adding square brackets around the URL.

Also every time you save, it adds more and more brackets.

```
// save 1
 * [1]: [https://www.google.com/]

// save 2
 * [1]: [[https://www.google.com/]]

// save 3
 * [1]: [[[https://www.google.com/]]]
```

There shouldn't be any brackets, the brackets are breaking the link.

**HOWEVER**

When I hover over the generated link in VS code, VS code seems to be interpreting it as a link to a local file.

So on top of the expectation above, this also needs to be considered:


**input**

```ts
/**
 * This is a [link][1] test.
 *
 * [1]: [local/path/to/file.js]
 *
 * @param value - Any value you want to test is a number.
 */
function test(value) {
    return true
}
```

**current output**

```ts
/**
 * This is a [link][1] test.
 *
 * [1]: [[local/path/to/file.js]]
 *
 * @param value - Any value you want to test is a number.
 */
function test(value) {
    return true
}
```

**expected output**

```ts
/**
 * This is a [link][1] test.
 *
 * [1]: [local/path/to/file.js]
 *
 * @param value - Any value you want to test is a number.
 */
function test(value) {
    return true
}
```

**Comment by hosseinmd** [comment]
Fixed v0.3.36
